bundles/org.openhab.binding.cloudrain/src/main/java/org/openhab/binding/cloudrain/internal/handler/CloudrainZoneHandler.java

```diff
@@ -0,0 +1,432 @@
+            this.remaining = remaining;
+            this.config = config;
+        }
+
+        @Override
+        public void run() {
+            if (--remaining >= 0) {
+                if (config.getRealtimeUpdates()) {
+                    updateState(channel.getUID(), new DecimalType(remaining));
+                }
+            } else {
+                // trigger thing update to reflect the end of the irrigation if desired
+                if (config.getUpdateAfterIrrigation()) {
+                    updateStatusFromAPI();
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates this CloudrainZoneHandler
+     *
+     * @param thing the Cloudrain Zone thing
+     * @param cloudrainAPI the initialized Cloudrain API. Handlers of this class expect that the
+     *            {@link CloudrainAccountHanlder} has already authenticated.
+     * @param timeZoneProvider the TimeZoneProvider for handling dates
+     * @param itemChannelLinkRegistry required for checking this Zone's channel link status
+     * @param timeZoneProvider a {@link TimeZoneProvider} for date time localization
+     */
+    public CloudrainZoneHandler(Thing thing, CloudrainAPI cloudrainAPI, TimeZoneProvider timeZoneProvider,
+            ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing);
+        this.cloudrainAPI = cloudrainAPI;
+        this.timeZoneProvider = timeZoneProvider;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+    }
+
+    @Override
+    public void initialize() {
+        // Retrieve this zone's ID from the properties as set by discovery
+        String zoneIdFromProperty = getThing().getProperties().get(PROPERTY_ZONE_ID);
+        if (zoneIdFromProperty != null) {
+            zoneId = zoneIdFromProperty;
+            // schedule initialization tasks
+            scheduler.execute(() -> {
+                CloudrainAccountHanlder handler = getAccountHanlder();
+                if (handler != null) {
+                    // Register the zone to receive property and status updates
+                    handler.registerZone(zoneId, getThing());
+                    // Call the Cloudrain API and update the thing's status now
+                    boolean successful = updateStatusFromAPI();
+                    if (successful) {
+                        // Register the zone to receive irrigation updates only if channels are linked
+                        // This is done to avoid unnecessary frequent API calls
+                        if (isAnyIrrigationChannelLinked()) {
+                            handler.registerForIrrigationUpdates(zoneId, getThing());
+                        }
+                        // Update the THING status
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                } else {
+                    logger.warn(ERROR_MSG_ZONE_REGISTRATION, zoneId);
+                    updateStatus(ThingStatus.OFFLINE);
```

Comment:
```suggestion
                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                    ERROR_MSG_ZONE_REGISTRATION +" Zone:"+zoneId);
```

Please use this format and choose the right ThingStatusDetail for each use case. Don't use the logger AND updateStatus as you only need one and the updateStatus is preferred as it is seen in the UI without needing to look at the logs...

Please go over and find and fix any other cases of this, thanks.

See the guidelines here:
https://www.openhab.org/docs/developer/guidelines.html#f-logging
and here:
https://www.openhab.org/docs/developer/bindings/#updating-the-thing-status

Reply:
Fixed